src/com/google/javascript/jscomp/ProcessCommonJSModules.java

```diff
@@ -222,6 +312,47 @@ private void visitScript(NodeTraversal t, Node script) {
       compiler.reportCodeChange();
     }
 
+    /**
+     * Rewrites CommonJS part of UMD pattern by removing the if-condition and the
+     * else-branch and adds the then-branch directly to the current parent node.
+     */
+    private void visitCommonJSIfStatement(Node n) {
+      Node p = n.getParent();
+      if (p != null) {
+        // pull out then-branch
+        replaceIfStatementWithBranch(n, n.getChildAtIndex(1));
+      }
+    }
+
+    /**
+     * Rewrites AMD part of UMD pattern by removing the if-condition and the
+     * then-branch and adds the else-branch directly to the current parent node.
+     */
+    private void visitAMDIfStatement(Node n) {
+      Node p = n.getParent();
+      if (p != null) {
+        if (n.getChildCount() == 3) {
+          // pull out else-branch
+          replaceIfStatementWithBranch(n, n.getChildAtIndex(2));
+        } else {
+          // remove entire if-statement if it doesn't have an else-branch
+          p.removeChild(n);
+        }
+      }
+    }
+
+    private void replaceIfStatementWithBranch(Node ifStatement, Node branch) {
+      Node p = ifStatement.getParent();
+      Node newNode = branch;
+      // remove redundant block node
+      if (branch.isBlock() && branch.getChildCount() == 1) {
```

Comment:
The output at the end looks the same, but the tests I've added would fail if we would remove this.